prune_shard_list.c

```diff
@@ -239,6 +249,51 @@ MakeOpExpression(Var *variable, int16 strategyNumber)
 
 
 /*
+ * LookupOperatorByType is a wrapper around GetOperatorByType that uses a cache
+ * to avoid multiple lookups of operators by their types.
```

Comment:
I'd say _multiple lookups of operators within a single session_, to make it clear that the cache is process-bound and not in shared memory.